Core/Core/Api/GraphQL/Client.GraphqlCleintOperations/Client.StreamOperations.cs

```diff
@@ -9,6 +9,48 @@ namespace Speckle.Core.Api;
 
 public partial class Client
 {
+
+  /// <summary>
+  /// Cheks if a stream exists by id.
+  /// </summary>
+  /// <param name="id">Id of the stream to get</param>
+  /// <returns></returns>
+  public Task<bool> StreamExists(string id)
+  {
+    return StreamExists(CancellationToken.None, id);
+  }
+
+  /// <summary>
+  /// Cheks if a stream exists by id.
+  /// </summary>
+  /// <param name="id">Id of the stream to get</param>
+  /// <returns></returns>
+  public async Task<bool> StreamExists(CancellationToken cancellationToken, string id)
+  {
+    try
+    {
+      var request = new GraphQLRequest
+      {
+        Query =
+          $@"query Stream($id: String!) {{
+                      stream(id: $id) {{
+                        id
+                      }}
+                    }}",
+        Variables = new { id }
+      };
+      var stream = (await ExecuteGraphQLRequest<StreamData>(request, cancellationToken).ConfigureAwait(false)).stream;
+
+      return stream.id == id;
+    }
+    catch
+    {
+      //ignore
```

Comment:
Do we need to swallow exceptions here?
The other client functions let Exceptions bubble up, no?

Reply:
makes sense, I gave it a stab!